### DISCUSSION TITLE
Long-lived instances of BleakClient/BleakScanner vs new instances

### DISCUSSION THREAD

**Discussion by x11x** [hbldh/bleak#1202] [Q&A]
I just wanted to check: the preferred pattern is to **create a new instance** every time you need to scan (`BleakScanner`), or connect to a device (`BleakClient`).

Especially with `BleakClient` this seems important due to inconsistent state in `BleakClient` after disconnect as mentioned in https://github.com/hbldh/bleak/issues/713#issuecomment-1005053102 and a few other places.

I'm not sure if this is also critical in `BleakScanner`, but most of the examples (such as this one https://github.com/hbldh/bleak/discussions/933#discussioncomment-3376981) use a new `BleakScanner` instance every time (the `BleakScanner.find_by_*` class methods create a short-lived `BleakScanner` instance internally).

So is it correct to say, its not really recommended to keep a `BleakScanner` instance and call `start()`, `stop()` and `start()` again, etc. as needed? Likewise with `BleakClient` it is not recommended to call `connect()` after a `disconnect()`?

Since bleak is a relatively young project, I wonder if it is a good idea for bleak to enforce the recommended pattern and just raise an exception when `connect()` is called after a previous `disconnect()` on a `BleakClient`, or `start()` after `stop()` on a `BleakScanner`? It might help with the OOP complexities of inconsistent state in long-lived instances mentioned in #713 and I think elsewhere.

Maybe this should have been an issue instead of a discussion post but I just wanted to check I have understood (happy to open an issue for this suggestion if anyone things it is worth it).

**Comment by dlech** [comment]
Starting and stopping a scanner multiple times should be fine.
